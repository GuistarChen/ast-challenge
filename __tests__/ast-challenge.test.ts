import generate from '@babel/generator';
import * as t from '@babel/types';
import MakeAst from "../src/index"
import Data from '../example-methods.json'

const name = Object.keys(Data)

console.log(Data)
const expectCode = (ast) => {
    console.log(ast)
    expect(
        generate(ast).code
    ).toMatchSnapshot();
}

const printCode = (ast) => {
    console.log(
        generate(ast).code
    );
}
//     it('works', () => {
//         let obj = {
//             "Pools": {
//                 "requestType": "QueryPoolsRequest",
//                 "responseType": "QueryPoolsResponse"
//             }
//         }
//      const key =   Object.keys(obj)
//      const value = Object.values(obj[key]) 
//     expectCode(MakeAst(`Use${key[0]}Query`,`use${key[0]}`,value[0],value[1],key[0].toLowerCase(),`${key[0].toLowerCase()}Query`))

// });
console.log(name)
name.map(n=>{
    console.log(n)
    const value = Object.values(Data[n]) 

    it('works', () => {

      expectCode(MakeAst(`Use${n}Query`,`use${n}`,value[0],value[1],n.toLowerCase(),`${n.toLowerCase()}Query`))

});
})