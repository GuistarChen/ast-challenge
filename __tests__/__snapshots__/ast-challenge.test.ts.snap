// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
"export interface UsePoolsQuery<TData> extends ReactQueryParams<QueryPoolsResponse, TData> {
  request: QueryPoolsRequest;
}

const usePools = ({
  request: request,
  options: options
}) => {
  return useQuery([\\"poolsQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.pools(request);
  }, options);
};"
`;

exports[`works 2`] = `
"export interface UseNumPoolsQuery<TData> extends ReactQueryParams<QueryNumPoolsResponse, TData> {
  request: QueryNumPoolsRequest;
}

const useNumPools = ({
  request: request,
  options: options
}) => {
  return useQuery([\\"numpoolsQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.numpools(request);
  }, options);
};"
`;

exports[`works 3`] = `
"export interface UseTotalLiquidityQuery<TData> extends ReactQueryParams<QueryTotalLiquidityResponse, TData> {
  request: QueryTotalLiquidityRequest;
}

const useTotalLiquidity = ({
  request: request,
  options: options
}) => {
  return useQuery([\\"totalliquidityQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.totalliquidity(request);
  }, options);
};"
`;

exports[`works 4`] = `
"export interface UsePoolQuery<TData> extends ReactQueryParams<QueryPoolResponse, TData> {
  request: QueryPoolRequest;
}

const usePool = ({
  request: request,
  options: options
}) => {
  return useQuery([\\"poolQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.pool(request);
  }, options);
};"
`;

exports[`works 5`] = `
"export interface UsePoolParamsQuery<TData> extends ReactQueryParams<QueryPoolParamsResponse, TData> {
  request: QueryPoolParamsRequest;
}

const usePoolParams = ({
  request: request,
  options: options
}) => {
  return useQuery([\\"poolparamsQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.poolparams(request);
  }, options);
};"
`;

exports[`works 6`] = `
"export interface UseTotalPoolLiquidityQuery<TData> extends ReactQueryParams<QueryTotalPoolLiquidityResponse, TData> {
  request: QueryTotalPoolLiquidityRequest;
}

const useTotalPoolLiquidity = ({
  request: request,
  options: options
}) => {
  return useQuery([\\"totalpoolliquidityQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.totalpoolliquidity(request);
  }, options);
};"
`;

exports[`works 7`] = `
"export interface UseTotalSharesQuery<TData> extends ReactQueryParams<QueryTotalSharesResponse, TData> {
  request: QueryTotalSharesRequest;
}

const useTotalShares = ({
  request: request,
  options: options
}) => {
  return useQuery([\\"totalsharesQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.totalshares(request);
  }, options);
};"
`;

exports[`works 8`] = `
"export interface UseSpotPriceQuery<TData> extends ReactQueryParams<QuerySpotPriceResponse, TData> {
  request: QuerySpotPriceRequest;
}

const useSpotPrice = ({
  request: request,
  options: options
}) => {
  return useQuery([\\"spotpriceQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.spotprice(request);
  }, options);
};"
`;

exports[`works 9`] = `
"export interface UseEstimateSwapExactAmountInQuery<TData> extends ReactQueryParams<QuerySwapExactAmountInResponse, TData> {
  request: QuerySwapExactAmountInRequest;
}

const useEstimateSwapExactAmountIn = ({
  request: request,
  options: options
}) => {
  return useQuery([\\"estimateswapexactamountinQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.estimateswapexactamountin(request);
  }, options);
};"
`;

exports[`works 10`] = `
"export interface UseEstimateSwapExactAmountOutQuery<TData> extends ReactQueryParams<QuerySwapExactAmountOutResponse, TData> {
  request: QuerySwapExactAmountOutRequest;
}

const useEstimateSwapExactAmountOut = ({
  request: request,
  options: options
}) => {
  return useQuery([\\"estimateswapexactamountoutQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.estimateswapexactamountout(request);
  }, options);
};"
`;
